### PR TITLE
 - ObjC Support  - MotionTransitionAnimationType

### PR DESCRIPTION
**Pull request by StefaniOSApps**
#70

### CODE REVIEW

File: Example/AppDelegate.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
```
Comment (daniel-jonathan) [inline Example/AppDelegate.h:1]: These comments will need to be removed and adding the MIT license we use in all other documents.

File: Example/AppDelegate.m [hunk 2]
```diff
@@ -0,0 +1,47 @@
+//
```
Comment (daniel-jonathan) [inline Example/AppDelegate.m:1]: Why are we adding an entire project here?

File: Example/ViewControllers.m [hunk 3]
```diff
@@ -0,0 +1,57 @@
+//
```
Comment (daniel-jonathan) [inline Example/ViewControllers.m:1]: Why is this class added?

**Review by daniel-jonathan — COMMENTED** [review]
Thank you for sending this, but you added all these extra files, as well, you have to remove all the top page declarations with your name, add MIT license like we have --- exactly.

**Review by OrkhanAlikhanov — COMMENTED** [review]
Hey! Thank you for the PR! I like your new enum approach! However, I don't think adding `navigationTransitionTypeObjC` solely for supporting objc is good decision. I'd like to think more on how to support obj-c

File: Example/AppDelegate.m (hunk 2 above)
Comment (StefaniOSApps) [reply Example/AppDelegate.m:1]: Its a UI to test this solution.

File: Example/ViewControllers.m (hunk 3 above)
Comment (StefaniOSApps) [reply Example/ViewControllers.m:1]: Its a UI to test this solution.

File: Example/AppDelegate.h (hunk 1 above)
Comment (StefaniOSApps) [reply Example/AppDelegate.h:1]: Its a UI to test this solution.

**Comment by StefaniOSApps** [comment]
It was the first solution I thought of. My solution uses the existing one. Of course It would be better not to write duplicate functions. I will add a PR later which no longer contains the Example UI. Before that, a solution must be found how to support MotionTransitionAnimationType in ObjC. If the solution is accepted by you, I will send you a new PR.

**Comment by StefaniOSApps** [comment]
@OrkhanAlikhanov The instance variable navigationTransitionTypeObjC is used to describe the instance variable navigationTransitionType via the setter. With ObjC you can not access navigationTransitionType directly.

**Comment by daniel-jonathan** [comment]
@StefaniOSApps yes, this is very much appreciated. I believe the next steps is to have @OrkhanAlikhanov and I figure out how we want to move forward in general, as @OrkhanAlikhanov mentioned with ObjC support. I 100% do not want to manage ObjC code in a Swift project. ObjC is a bonus in my mind with these frameworks as we are moving with Apple's approach of Swift across the board.

**Comment by StefaniOSApps** [comment]
I have now replace the instance var by a func. Do you find this solution better? Of course, names used can still be changed, and the sample files are then also removed.

**Comment by daniel-jonathan** [comment]
@StefaniOSApps thank you! We will review this and on Monday we will have an answer going forward. We appreciate your work :)

**Comment by adamdahan** [comment]
@StefaniOSApps Unfortunately, it is an internal decision to no longer support Objc in our upcoming updates for Material 4. I do apologize for the inconvenience this has caused you & your application, remember, you can always keep your own custom version of Material locally with your objc patches. 

Please open a new ticket if there is anything else we can help you with :)

**Pull request closed without merging**